src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const arr = people.filter(person =>
+    century
+      ? person.sex === 'm'
+      && Math.ceil(person.died / 100) === century
+      : person.sex === 'm');
+
+  const ages = arr.map(person => person.died - person.born);
+
+  function getAvarageAge() {
+    return ages.reduce((sum, age) => sum + age) / ages.length;
+  }
```

Comment:
```suggestion
  function getAvarageAge(people) {
    return people.reduce((sum, person) => sum + person.died - person.born) / people.length;
  }
```

move this function to uuper scope, that will help you to use one function in the first and second function instead of creating duplicates

Reply:
create a separate helper function to calculate average age outside of all your main functions. this function should take array with people, and return average age of these people